### PR TITLE
Populate entity trait reference from the parent entity.

### LINKED ISSUE #6: Populate Entity trait reference from Entity
Prevent user from editing Entity trait reference fields (Customer, Salesperson, Vendor) and populate from the Entity during trait creation.

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**
Closes #6, #7

### CODE REVIEW

**Comment by michaelachrisco** [comment]
@josephbridgwaterrowe Reviewing now.

File: spec/interactors/create_salesperson_spec.rb [hunk 1]
```diff
@@ -3,36 +3,27 @@
...
-                 :model => I18n.t('ar.models.salesperson')))
-      end
-
-      it 'should have salesperson present' do
-        expect(context.salesperson).to be_present
-      end
+    describe Interactor::Context do
+      it { is_expected.to be_a_success }
+      # rubocop:disable Metrics/LineLength
+      its(:message) { is_expected.to eq(I18n.t('app.salesperson.create.success')) }
```
Comment (michaelachrisco) [inline spec/interactors/create_salesperson_spec.rb]: You can split this up as: 
```Ruby
        its(:message) do
          is_expected.to eq(I18n.t('app.salesperson.create.success'))
        end
```

File: spec/interactors/create_salesperson_spec.rb [hunk 2]
```diff
@@ -70,20 +62,14 @@
...
-
-      it 'has set the salesperson' do
-        expect(context.salesperson).to be_present
-      end
+    let(:salesperson_entity) { nil }
+
+    describe Interactor::Context do
+      it { is_expected.to be_a_failure }
+      # rubocop:disable Metrics/LineLength
+      its(:message) { is_expected.to eq(I18n.t('app.salesperson.create.invalid')) }
```
Comment (michaelachrisco) [inline spec/interactors/create_salesperson_spec.rb]: Same here.

**Comment by michaelachrisco** [comment]
Very small changes but good to go otherwise.

File: spec/interactors/create_salesperson_spec.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline spec/interactors/create_salesperson_spec.rb]: I know, but to be honest I prefer the one line version... granted they both take 3 lines :)

Comment (michaelachrisco) [inline spec/interactors/create_salesperson_spec.rb]: Well its a small change either way. Merging.

**Pull request merged**